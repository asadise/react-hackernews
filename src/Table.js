import React, { Component } from "react";

function isSearched(searchTerm) {
    return item => item.title.toLowerCase().includes(searchTerm.toLowerCase());
}

class Table extends Component {
    render() {
        const { list, pattern, onDismiss } = this.props;
        return (
            <div className="table">
                {/*{this.state.list.filter(isSearched(this.state.searchTerm)).map((item, index) =>  */}
                {list.filter(isSearched(pattern)).map((item, index) =>
                    <div key={item.id} className="table-row">
                        <span style={{ width: '10%' }}>
                            <img width="30" alt="cover" src={item.image} />
                        </span>
                        <span style={{ width: '30%' }}>
                            {index}. <a href={item.url}>{item.title.split("-")[0]}</a>
                        </span>
                        <span style={{ width: '20%' }}>{item.title.split("-")[1]}</span>
                        <span style={{ width: '10%' }}> 4</span>
                        <span style={{ width: '10%' }}> 7</span>
                        <Button className="button-inline" onClick={() => onDismiss(item.id)}>حذف</Button>
                        <br /><br />
                    </div>
                )}
            </div>
        );
    }
}

/*class Button extends Component {
    render() {
        const { onClick, className = '', children } = this.props;
        return (<button onClick={onClick} className={className} type="button">{children}</button>);
    }
}*/
const Button = ({ onClick, className = '', children }) => <button onClick={onClick} className={className} type="button">{children}</button>;


export default Table;