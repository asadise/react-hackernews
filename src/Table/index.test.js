import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Table from '../Table';
//import Button from '../Table';

describe('Table', () => {
  const props = {
    list: [
      { id: 0, title: 'test1', url: 'https://test-url1.com', image: '1.jpg' },
      { id: 1, title: 'test2', url: 'https://test-url2.com', image: '2.jpg' },
    ], page: 1,
  };
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(<Table {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/*describe('Button', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Give me More!</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(<Button>Give me More!</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
}); */