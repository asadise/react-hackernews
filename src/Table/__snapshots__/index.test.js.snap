// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has a valid snapshot 1`] = `
<div>
  <div
    className="table"
  >
    <div
      className="table-row"
    >
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <img
          alt="cover"
          src="1.jpg"
          width="30"
        />
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        0
        . 
        <a
          href="https://test-url1.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          test1
        </a>
      </span>
      <span
        style={
          Object {
            "width": "20%",
          }
        }
      />
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
         4
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
         7
      </span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        حذف
      </button>
      <br />
      <br />
    </div>
    <div
      className="table-row"
    >
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <img
          alt="cover"
          src="2.jpg"
          width="30"
        />
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        1
        . 
        <a
          href="https://test-url2.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          test2
        </a>
      </span>
      <span
        style={
          Object {
            "width": "20%",
          }
        }
      />
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
         4
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
         7
      </span>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        حذف
      </button>
      <br />
      <br />
    </div>
  </div>
  <div
    className="interactions"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      بارگزاری موارد بیشتر...
    </button>
  </div>
</div>
`;
